voluncheer/opportunityboard/views/postanopportunity.py

```diff
@@ -13,15 +17,20 @@ def post_an_opportunity(request):
     if user.is_organization:
         organization_profile = Organization.objects.get(pk=user)
     if request.method == "POST":
-        form = PostAnOpportunityForm(request.POST, request.FILES, instance=request.user)
+        # Set Organization based on current user
+        post = request.POST.copy()
+        post.update({"organization": organization_profile})
```

Comment:
Why do you have to copy and then update? Why can't you directly update?

Reply:
Because the organization field is hidden in the form. Users shouldn't have to specify which organization they are, it should happen automatically in the code. This relates to above change (removing `instance=request.user`)

Reply:
So you cannot do: request.POST.update({"organization": organization_profile}) ?

Reply:
My understanding is you have to copy `request.POST` to make it mutable, but I'll double check

Reply:
I confirmed you have to make a copy before updating it. I agree it's weird, but necessary.